v2/internal/controllers/to_azure_secrets_test.go

```diff
@@ -66,10 +69,121 @@ func Test_MissingSecretKey_ReturnsError(t *testing.T) {
 
 	tc.CreateResourceAndWaitForState(vm, metav1.ConditionFalse, conditions.ConditionSeverityWarning)
 	// We expect the ready condition to include details of the error
-	tc.Expect(vm.Status.Conditions[0].Reason).To(Equal("SecretNotFound"))
+	tc.Expect(vm.Status.Conditions[0].Reason).To(Equal(conditions.ReasonSecretNotFound))
 	tc.Expect(vm.Status.Conditions[0].Message).To(
 		ContainSubstring("Secret \"%s/%s\" does not contain key \"%s\"", tc.Namespace, secret.Name, secret.Key))
 
 	// Delete VM and resources.
 	tc.DeleteResourcesAndWait(vm, networkInterface, subnet, vnet, rg)
 }
+
+func Test_UserSecretInDifferentNamespace_SecretNotFound(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+	secretName := tc.Namer.GenerateName("admin-pw")
+	nameSpaceName := "secret-namespace"
+
+	createNamespaces(tc, nameSpaceName)
+	secretInDiffNamespace := createNamespacedSecret(tc, nameSpaceName, secretName)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	vnet := newVMVirtualNetwork(tc, testcommon.AsOwner(rg))
+	subnet := newVMSubnet(tc, testcommon.AsOwner(vnet))
+	networkInterface := newVMNetworkInterface(tc, testcommon.AsOwner(rg), subnet)
+
+	tc.CreateResourceAndWait(vnet)
+	tc.CreateResourceAndWait(subnet)
+	tc.CreateResourceAndWait(networkInterface)
+
+	vm := newVM(tc, rg, networkInterface, secretInDiffNamespace)
+
+	tc.CreateResourceAndWaitForState(vm, metav1.ConditionFalse, conditions.ConditionSeverityWarning)
+	// We expect the ready condition to include details of the error
+	tc.Expect(vm.Status.Conditions[0].Reason).To(Equal(conditions.ReasonSecretNotFound))
+	tc.Expect(vm.Status.Conditions[0].Message).To(
+		ContainSubstring("failed resolving secret references: %s/%s does not exist", tc.Namespace, secretInDiffNamespace.Name))
+
+	// Delete VM and resources.
+	tc.DeleteResourcesAndWait(vm, networkInterface, subnet, vnet, rg)
+}
+
+func Test_UserSecretInDifferentNamespace_ShouldNotTriggerReconcile(t *testing.T) {
+	t.Parallel()
+	ns1 := "ns-1"
+	ns2 := "ns-2"
+
+	cfg, err := config.ReadFromEnvironment()
+	if err != nil {
+		t.Fatal(err)
+	}
+	cfg.TargetNamespaces = []string{ns1, ns2}
+	cfg.SyncPeriod = nil
```

Comment:
Rather than putting this here, can we do one of two things?

1. Change `ForTestWithConfig` to force this parameter to be `nil` always? `ForTest` does it right now so you could just move that.
2. Make a test specific `ReadFromEnvironmentForTest()` that forces it to be nil. 

The small advantage of 2 is that it means if you really wanted to leave this value set you can actually do it by calling `ReadFromEnvironment` instead of the test one. As part of doing 2 you'd wanna move all the calls to `ReadFromEnvironment` to `ReadFromEnvironmentForTest` in the test package.

Reply:
done for this and all below!